src/main/java/org/spongepowered/api/text/dialogue/AnswerProcessor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.dialogue;
+
+import java.util.Optional;
+
+/**
+ * A functional interface which processes {@link Answer}s.
+ */
+@FunctionalInterface
+public interface AnswerProcessor {
+
+    /**
+     * Processes an {@link Answer} and returns the next {@link Question} in
+     * the {@link Dialogue}.
+     *
+     * @param dialogue The current dialogue
+     * @param question The question being answered
+     * @param answer The answer to the question
+     * @return The next {@link Question} in the {@link Dialogue}. If {@link
+     * Optional#empty()}, the dialogue is concluded.
+     */
+    Optional<Question> process(Dialogue dialogue, Question question, Answer answer);
```

Comment:
The answer already contains references to both the Dialogue and the Question.  IMO this would be better simplified to just accepting an Answer.